libraries/core_libs/network/src/tarcap/packets_handlers/transaction_packet_handler.cpp

```diff
@@ -24,31 +24,33 @@ inline void TransactionPacketHandler::process(const PacketData &packet_data, con
   std::string received_transactions;
   const auto transaction_count = packet_data.rlp_.itemCount();
 
-  SharedTransactions transactions;
+  std::vector<std::pair<std::shared_ptr<Transaction>, TransactionStatus>> transactions;
   transactions.reserve(transaction_count);
 
   for (size_t tx_idx = 0; tx_idx < transaction_count; tx_idx++) {
-    std::shared_ptr<Transaction> transaction;
+    std::pair<std::shared_ptr<Transaction>, TransactionStatus> transaction;
 
     try {
-      transaction = std::make_shared<Transaction>(packet_data.rlp_[tx_idx].data().toBytes());
+      transaction.first = std::make_shared<Transaction>(packet_data.rlp_[tx_idx].data().toBytes());
     } catch (const Transaction::InvalidSignature &e) {
       throw MaliciousPeerException("Unable to parse transaction: " + std::string(e.what()));
     }
 
     if (dag_blk_mgr_) [[likely]] {  // ONLY FOR TESTING
```

Comment:
if dag_blk_mgr == nullptr (tests), you dont set transaction.second - is that ok ?

Reply:
Fixed

Reply:
should this `transaction.second = TransactionStatus::Verified;` be outside of `if (dag_blk_mgr_)` to acutally fix it ?

Reply:
Fixed